programs/token-2022/src/state/multisig.rs

```diff
@@ -0,0 +1,93 @@
+use pinocchio::{
+    account_info::{AccountInfo, Ref},
+    program_error::ProgramError,
+    pubkey::Pubkey,
+};
+
+use crate::{instructions::MAX_MULTISIG_SIGNERS, ID};
+
+/// Multisignature data.
+#[repr(C)]
+pub struct Multisig {
+    /// Number of signers required
+    pub m: u8,
+    /// Number of valid signers
+    pub n: u8,
+    /// Is `true` if this structure has been initialized
+    pub is_initialized: u8,
+    /// Signer public keys
+    pub signers: [Pubkey; MAX_MULTISIG_SIGNERS],
+}
+
+impl Multisig {
+    /// The length of the `Multisig` account data.
+    pub const LEN: usize = core::mem::size_of::<Multisig>();
+
+    /// Return a `Multisig` from the given account info.
+    ///
+    /// This method performs owner and length validation on `AccountInfo`, safe borrowing
+    /// the account data.
+    #[inline]
+    pub fn from_account_info(account_info: &AccountInfo) -> Result<Ref<Multisig>, ProgramError> {
+        if account_info.data_len() != Self::LEN {
+            return Err(ProgramError::InvalidAccountData);
+        }
+        if !account_info.is_owned_by(&ID) {
+            return Err(ProgramError::InvalidAccountOwner);
+        }
+        Ok(Ref::map(account_info.try_borrow_data()?, |data| unsafe {
+            Self::from_bytes_unchecked(data)
+        }))
+    }
+
+    /// Return a `Multisig` from the given account info.
+    ///
+    /// This method performs owner and length validation on `AccountInfo`, but does not
+    /// perform the borrow check.
+    ///
+    /// # Safety
+    ///
+    /// The caller must ensure that it is safe to borrow the account data (e.g., there are
+    /// no mutable borrows of the account data).
+    #[inline]
+    pub unsafe fn from_account_info_unchecked(
+        account_info: &AccountInfo,
+    ) -> Result<&Self, ProgramError> {
+        if account_info.data_len() != Self::LEN {
+            return Err(ProgramError::InvalidAccountData);
+        }
+        if account_info.owner() != &ID {
+            return Err(ProgramError::InvalidAccountOwner);
+        }
+        Ok(Self::from_bytes_unchecked(
+            account_info.borrow_data_unchecked(),
+        ))
+    }
+
+    /// Return a `Multisig` from the given bytes.
+    ///
+    /// # Safety
+    ///
+    /// The caller must ensure that `bytes` contains a valid representation of `Multisig`, and
+    /// it is properly aligned to be interpreted as an instance of `Multisig`.
+    /// At the moment `Multisig` has an alignment of 1 byte.
+    /// This method does not perform a length validation.
+    #[inline(always)]
+    pub unsafe fn from_bytes_unchecked(bytes: &[u8]) -> &Self {
+        &*(bytes.as_ptr() as *const Multisig)
+    }
+
+    #[inline(always)]
+    pub fn required_signers(&self) -> u8 {
+        self.m
+    }
+
+    pub fn valid_signers(&self) -> u8 {
+        self.n
+    }
```

Comment:
More convenient version:
```suggestion
    #[inline(always)]
    pub const fn len(&self) -> usize {
        self.n as usize
    }
```